lib/Listener/ChattyLLMTaskListener.php

```diff
@@ -61,6 +61,7 @@ public function handle(Event $event): void {
 			$message->setRole('assistant');
 			$message->setContent(trim($task->getOutput()['output'] ?? ''));
 			$message->setTimestamp(time());
+			$message->setSources(json_encode($task->getOutput()['sources'] ?? []));
```

Comment:
json_encode returns false on failure, we might want to check for that

Reply:
```suggestion
			$sources = json_encode($task->getOutput()['sources'] ?? []);
			$message->setSources($sources ? $sources : '[]');
```

I think we want this